sql-plugin/src/main/scala/com/nvidia/spark/rapids/spill/SpillFramework.scala

```diff
@@ -177,14 +177,47 @@ trait SpillableHandle extends StoreHandle {
    */
   def spill(): Long
 
+  private var spilling = false
+  private val spillLock = new Object()
```

Comment:
It appears we could simplify this design by using AtomicBoolean which supports CAS on the value to ensure only one thread can successfully set it to true.

Reply:
agree if we can make it simpler with CAS operations it should be worth trying that.